src/WebAPI/Controllers/PlayerController.cs

```diff
@@ -1,48 +1,57 @@
+using MediatR;
 using Microsoft.AspNetCore.Mvc;
-using NeoServer.Web.API.Services.Interfaces;
-using NeoServer.Web.Shared.ViewModels.Request;
+using NeoServer.Web.API.Requests.Commands;
+using NeoServer.Web.API.Requests.Queries;
+using NeoServer.Web.API.Response.Constants;
 
 namespace NeoServer.Web.API.Controllers;
 
 [ApiController]
 [Route("api/[controller]")]
-public class PlayerController : BaseController
+public class PlayerController(IMediator mediator) : BaseController
 {
-    #region private members
-
-    private readonly IPlayerApiService _playerService;
-
-    #endregion
-
-    #region constructor
+    [HttpGet]
+    public async Task<IActionResult> GetAllAsync([FromQuery] GetPlayersRequest request)
+       => Ok(await mediator.Send(request));
 
-    public PlayerController(IPlayerApiService playerService)
+    [HttpGet("{id}")]
+    public async Task<IActionResult> GetById([FromRoute] int id)
     {
-        _playerService = playerService;
+        var response = await mediator.Send(new GetPlayerByIdRequest { Id = id });
+        if (response is null) return NotFound();
+        
+        return Ok(response);
     }
 
-    #endregion
-
-    #region public methods
-
-    [HttpGet]
-    public async Task<IActionResult> GetAllAsync()
+    [HttpPost]
+    public async Task<IActionResult> Post([FromBody] CreatePlayerRequest request)
     {
-        return Response(await _playerService.GetAll());
+        var response = await mediator.Send(request);
+        if (!response.IsSuccess) 
+            return UnprocessableEntity(response.ErrorMessage);
+        
+        return Ok(SuccessMessage.PlayerCreated.Replace("{id}", response.Identifier.ToString()));
     }
-
-    [HttpGet("{playerId}")]
-    public async Task<IActionResult> GetById([FromRoute] int playerId)
+    
+    [HttpPatch("{id}/skills")]
+    public async Task<IActionResult> UpdateSkills(int id, [FromBody] UpdatePlayerSkillsRequest request)
     {
-        return Response(await _playerService.GetById(playerId));
+        request.SetPlayerId(id);
+        var response = await mediator.Send(request);
+        if (!response.IsSuccess) 
+            return UnprocessableEntity(response.ErrorMessage);
+        
+        return Ok(SuccessMessage.PlayerSkillsUpdated);
     }
-
-    [HttpPost]
-    public async Task<IActionResult> Post([FromBody] PlayerPostRequest player)
+    
+    [HttpPatch("{id}/infos")]
```

Comment:
lets rename this endpoint to something more meaningful in future. for now it is good

Reply:
This endpoint name was based the admin of tibia.
![image](https://github.com/user-attachments/assets/d8edc18d-0b37-4af8-9f7e-fe0a0b3679fc)